datamodel/core.py

```diff
@@ -378,17 +374,7 @@ def __init__(self, env, registry):
         self.registry = registry
 
     def __getitem__(self, key):
-        model = self.registry[key]
-        model.__init__ = functools.partialmethod(model.__init__, env=self.env)
-
-        @classmethod
-        def __get_schema_class__(cls, **kwargs):
-            cls = cls.__schema_class__(**kwargs)
-            cls._env = self.env
```

Comment:
When the error was raised, this was being referenced every time, this showed that something wrong happened here.

Once found, the only option was trying to remove all dependencies on this.